### PR TITLE
Fixes 136 - Update CSS styleguide

### PR DESCRIPTION
**Pull request by goodwinchris**

### CODE REVIEW

File: ux-dev/css.md [hunk 1]
```diff
@@ -1,78 +1,82 @@
...
+* Firefox
+* Safari
+* IE/Edge
+
+As a result, on most projects we can use Flexbox and/or Grid.
+
+Always be sure to double check with an individual client to confirm browser support.
+
+## General File Structure
+Though some projects may have different requirements, DockYard generally has set of starter files such as:
```
Comment (Ctannerweb) [inline ux-dev/css.md]: We could mention the modules folder with a module-foo.css file within it

**Comment by jsit** [comment]
I notice you're removing the property declaration order section -- should we include a link to our `stylelint-narwin-config` file for property order?

https://github.com/DockYard/stylelint-config-narwin/blob/master/index.js

**Comment by azzcatDY** [comment]
Looking good!

**Comment by goodwinchris** [comment]
We need to update to unitless zero values in this doc

File: ux-dev/css.md [hunk 2]
```diff
@@ -1,78 +1,90 @@
...
+* Each declaration should appear on its own line for more accurate error reporting.
+* End all declarations with a semi-colon.
+* Comma-separated property values should include a space after each comma (e.g., box-shadow).
+* Don't include spaces after commas within rgb(), rgba(), hsl(), hsla(), or rect() values. This helps differentiate multiple color values (comma, no space) from multiple property values (comma with space).
+* Don't prefix property values or color parameters with a leading zero (e.g., .5 instead of 0.5 and -.5px instead of -0.5px).
+* Lowercase all hex values, e.g., #fff. Lowercase letters are much easier to discern when scanning a document as they tend to have more unique shapes.
+* Use shorthand hex values where available, e.g., #fff instead of #ffffff.
+* Quote attribute values in selectors, e.g., input[type="text"]. They’re only optional in some cases, and it’s a good practice for consistency.
+* Always use double quotes in selectors and single quotes in the declaration
+* Always specify units e.g., margin: 0px; instead of margin: 0;.
```
Comment (monicawheeler) [inline ux-dev/css.md]: Is there a reason we are adding units to a zero value? This seems confusing. What if I put 0%? or 0rem? When would I put `px` over another unit on zero values? 

It seems clearer to not use units on zero values.

File: ux-dev/css.md [hunk 3]
```diff
@@ -1,78 +1,90 @@
...
+* Pseudo element should always have a double semicolon, e.g., .element::after
+* Nested selectors should have a space between each block:
```
Comment (monicawheeler) [inline ux-dev/css.md]: I think there are exceptions to this. Could we have a visual here?

File: ux-dev/css.md (hunk 2 above)
Comment (goodwinchris) [reply ux-dev/css.md]: oops I skimmed the post to check against this comment from Jan 24 and missed this. I'll update here.

File: ux-dev/css.md (hunk 3 above)
Comment (goodwinchris) [reply ux-dev/css.md]: Sure, what do you see as the exceptions?

File: ux-dev/css.md [hunk 4]
```diff
@@ -1,78 +1,90 @@
...
+* Chrome
+* Firefox
+* Safari
+* IE/Edge
+
+As a result, on most projects we can use Flexbox and/or Grid.
+
+Always be sure to double check with an individual client to confirm browser support.
+
+## General File Structure
```
Comment (monicawheeler) [inline ux-dev/css.md]: Awesome section!

File: ux-dev/css.md [hunk 5]
```diff
@@ -1,78 +1,90 @@
...
+Most projects should have our [style linter](https://github.com/DockYard/stylelint-config-narwin) installed. The linter helps us to define and maintain an [order of declaration](https://github.com/DockYard/stylelint-config-narwin/blob/master/index.js#L1).
+
+You can also check out the [Linter Styleguide](https://github.com/dockyard/styleguides/blob/master/ux-dev/stylelint-config-narwin.md).
+
+## Variables
+Narwin-pack uses [postcss-custom-properties](https://github.com/postcss/postcss-custom-properties).
+
+Most projects will have a variables file that holds all your global variables. Though there may be times when it makes sense to add variables to individual files, as a general rule variables should live in the variables.css file.
+
+## Media queries
```
Comment (monicawheeler) [inline ux-dev/css.md]: There are various ways to write a media query, for example: `@media only screen and (){}` - we could lay out that we use `@media () {}` and why.

File: ux-dev/css.md [hunk 6]
```diff
@@ -1,78 +1,90 @@
...
+* Include one space after : for each declaration.
+* Each declaration should appear on its own line for more accurate error reporting.
+* End all declarations with a semi-colon.
+* Comma-separated property values should include a space after each comma (e.g., box-shadow).
+* Don't include spaces after commas within rgb(), rgba(), hsl(), hsla(), or rect() values. This helps differentiate multiple color values (comma, no space) from multiple property values (comma with space).
+* Don't prefix property values or color parameters with a leading zero (e.g., .5 instead of 0.5 and -.5px instead of -0.5px).
+* Lowercase all hex values, e.g., #fff. Lowercase letters are much easier to discern when scanning a document as they tend to have more unique shapes.
+* Use shorthand hex values where available, e.g., #fff instead of #ffffff.
+* Quote attribute values in selectors, e.g., input[type="text"]. They’re only optional in some cases, and it’s a good practice for consistency.
+* Always use double quotes in selectors and single quotes in the declaration
```
Comment (monicawheeler) [inline ux-dev/css.md]: The linter calls for string quotes to be double. Should we update this syntax requirement?

File: ux-dev/css.md [hunk 7]
```diff
@@ -1,78 +1,95 @@
...
+We are currently using [PostCSS](https://github.com/postcss/postcss/) with [narwin-pack](https://github.com/dockyard/narwin-pack).  However, some of our older projects are still written in Sass/SCSS.
+
+## Syntax
+
+* Use soft tabs with two spaces—they're the only way to guarantee code renders the same in any environment.
+* When grouping selectors, each selector goes on a new line
+* Include one space before the opening brace of declaration blocks for legibility.
+* Place closing braces of declaration blocks on a new line.
+* Include one space after : for each declaration.
+* Each declaration should appear on its own line for more accurate error reporting.
```
Comment (philschanely) [inline ux-dev/css.md]: You may have this in mind as acceptable, but isfor things like `grid-template-areas` where properties are beneficial read this way:
```css
.thing {
  grid-template-areas:
    "head    head    help"
    "content content help"
    "foot    foot    foot";
}
```
Others properties may benefit from this as well. Perhaps the wording here could be, "Declarations should be place on separate lines for more accurate error reporting; some declarations such as `grid-template-areas` may span multiple lines if needed)." Or something like that?

File: ux-dev/css.md [hunk 8]
```diff
@@ -1,78 +1,95 @@
...
+* End all declarations with a semi-colon.
+* Comma-separated property values should include a space after each comma (e.g., box-shadow).
+* Don't include spaces after commas within rgb(), rgba(), hsl(), hsla(), or rect() values. This helps differentiate multiple color values (comma, no space) from multiple property values (comma with space).
+* Don't prefix property values or color parameters with a leading zero (e.g., .5 instead of 0.5 and -.5px instead of -0.5px).
+* Lowercase all hex values, e.g., #fff. Lowercase letters are much easier to discern when scanning a document as they tend to have more unique shapes.
+* Use shorthand hex values where available, e.g., #fff instead of #ffffff.
+* Quote attribute values in selectors, e.g., input[type="text"]. They’re only optional in some cases, and it’s a good practice for consistency.
+* Always use double quotes in selectors
+* Always specify units unless that value is 0 e.g., margin: 12px; instead of margin: 12; and margin: 0; instead of margin: 0px;.
```
Comment (philschanely) [inline ux-dev/css.md]: Do we have any guidance or best practices around which units we prefer for different things? I.E. do we try to use `rem` for type?

File: ux-dev/css.md [hunk 9]
```diff
@@ -1,78 +1,95 @@
...
+	* Holds all `t-...` styles for typography
+* `variables.css`
+	* Holds all global variables for a project
+* `fonts.css`
+	* Holds all `@font-face` declarations
+* `layout.css`
+	* Holds all global `l-…` styles
+* `modules/`
+  * Holds all app specific module files:
+    * `modules/search-bar.css`, `modules/cards.css`
```
Comment (philschanely) [inline ux-dev/css.md]: Do we ever allow subfolders inside `modules/`? If so, are there any rules around folder name vs. file name?

File: ux-dev/css.md [hunk 10]
```diff
@@ -1,78 +1,95 @@
...
+Most projects should have our [style linter](https://github.com/DockYard/stylelint-config-narwin) installed. The linter helps us to define and maintain an [order of declaration](https://github.com/DockYard/stylelint-config-narwin/blob/master/index.js#L1).
+
+You can also check out the [Linter Styleguide](https://github.com/dockyard/styleguides/blob/master/ux-dev/stylelint-config-narwin.md).
+
+## Variables
+Narwin-pack uses [postcss-custom-properties](https://github.com/postcss/postcss-custom-properties).
+
+Most projects will have a variables file that holds all your global variables. Though there may be times when it makes sense to add variables to individual files, as a general rule variables should live in the variables.css file.
+
+## Media queries
```
Comment (crowjm) [inline ux-dev/css.md]: Would it be useful to mention the way in which we target screens inside media queries? Max-width, min-width & max-width, and then min-wdth? Or is that only standard practice on Prodicle projects?

Comment (goodwinchris) [reply ux-dev/css.md]: I'm down to add it if we do it everywhere. I think thats how we do it on DY.com

File: ux-dev/css.md [hunk 11]
```diff
@@ -1,78 +1,95 @@
...
+
+## Order of Declaration
+Most projects should have our [style linter](https://github.com/DockYard/stylelint-config-narwin) installed. The linter helps us to define and maintain an [order of declaration](https://github.com/DockYard/stylelint-config-narwin/blob/master/index.js#L1).
+
+You can also check out the [Linter Styleguide](https://github.com/dockyard/styleguides/blob/master/ux-dev/stylelint-config-narwin.md).
+
+## Variables
+Narwin-pack uses [postcss-custom-properties](https://github.com/postcss/postcss-custom-properties).
+
+Most projects will have a variables file that holds all your global variables. Though there may be times when it makes sense to add variables to individual files, as a general rule variables should live in the variables.css file.
```
Comment (philschanely) [inline ux-dev/css.md]: Do we have any guidance or precedents for naming conventions for variables? `color-grey` vs. just `grey`, etc.?

File: ux-dev/css.md (hunk 9 above)
Comment (goodwinchris) [reply ux-dev/css.md]: I don't think we've done this but I don't think we have a super hard rule against it if we had a good reason to do it either. I'd lean towards not saying anything.

File: ux-dev/css.md (hunk 11 above)
Comment (goodwinchris) [reply ux-dev/css.md]: I'm not sure, it'd likely depend on the project.

**Comment by philschanely** [comment]
Good to go from me! 👍

File: ux-dev/css.md [hunk 12]
```diff
@@ -1,78 +1,104 @@
 # CSS
+## General info
+We are currently using [PostCSS](https://github.com/postcss/postcss/) with [narwin-pack](https://github.com/dockyard/narwin-pack).  However, some of our older projects are still written in Sass/SCSS.
+
+## Syntax
+
+* Use soft tabs with two spaces—they're the only way to guarantee code renders the same in any environment.
```
Comment (danrichman) [inline ux-dev/css.md]: FWIW, this is supposed to be handled by `.editorconfig`
https://github.com/dockyard/styleguides/blob/master/ux-dev/ux-developer-onboarding.md#editors

Comment (goodwinchris) [reply ux-dev/css.md]: Good call, done!

**Pull request merged**